contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -277,9 +288,27 @@ Need to remove inheritance to fix it.
             if($result.ToLower().Startswith('y'))
             {   
                 $needChange = $true
-                $sshAce = New-Object System.Security.AccessControl.FileSystemAccessRule `
-                    ($a.IdentityReference, "Read", "None", "None", "Allow")
-                $acl.SetAccessRule($sshAce)
+                $idRefShortValue = ($a.IdentityReference.Value).split('\')[-1]
+                if ($idRefShortValue -in $specialIdRefs )
+                {
+                    $ruleIdentity = Get-UserSID -User (New-Object Security.Principal.NTAccount $idRefShortValue)
```

Comment:
since this logic is duplicated in two places.. better to create a internal method so that it can be expanded in future as and when we run into this kind of exceptions... Its easy to maintain if its in one place..